axelrod/strategies/darwin.py

```diff
@@ -23,7 +23,14 @@ class Darwin(Player):
     """
 
     name = "Darwin"
-    memory_depth = float('inf')
+    behaviour = {
```

Comment:
That's the point: it would not. Not sure if that differentiation is needed? 

I'm suggesting that to deal with the issue raised by @marcharper with `Darwin` which is the only strategy that cheats by modifying it's own state (it doesn't reset properly)...